### ISSUE TITLE
Azure DevOps Build pipeline - error caused by warning messages

### ISSUE THREAD

**Issue by makuhn** [microsoft/navcontainerhelper#585]
I get following error when using `Publish-NewApplicationToNavContainer` in a build pipeline:
```
##[error]Where-Object : The property 'Name' cannot be found on this object. Verify that the property exists.
##[error]At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.6.3.1\AppHandling\Clean-BcContainerDatabase.ps1:38 

##[error]char:127

##[error]+ ...  DependenciesLast | Where-Object { $_.Name -ne "System Application" }

##[error]+                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

##[error]    + CategoryInfo          : NotSpecified: (:) [Where-Object], PropertyNotFoundException

##[error]    + FullyQualifiedErrorId : PropertyNotFoundStrict,Microsoft.PowerShell.Commands.WhereObjectCommand

##[error] 

##[error]PowerShell exited with code '1'.
```
Statement that causes the error:
```
Publish-NewApplicationToNavContainer -containerName $containerName `
                                 -appDotNetPackagesFolder (Join-Path $baseAppFolder ".netpackages") `
                                 -appFile $app `
                                 -credential $credential `
                                 -useCleanDatabase `
                                 -saveData
```
I analyzed the error and modified the Cmdlet in order to get the content of following statement:
```
    #$installedApps = Get-NavContainerAppInfo -containerName $containerName -tenantSpecificProperties -sort DependenciesLast | Where-Object { $_.Name -ne "System Application" }
    $installedApps = Get-NavContainerAppInfo -containerName $containerName -tenantSpecificProperties -sort DependenciesLast
    Write-Host "Installed apps:"
    $installedApps
    $installedApps = $installedApps | Where-Object { $_.Name -ne "System Application" }
```
Then I get following output in pipeline:
```
Installed apps:
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.
WARNING: Caution: Your program license expires in 14 days.
WARNING: This license is not compatible with this version of Business Central.


ServerInstance       : MicrosoftDynamicsNavServer$BC
ExtensionDataVersion : 15.0.35178.0
IsInstalled          : True
SyncState            : Synced
NeedsUpgrade         : False
IsPublished          : True
...
```
Variable `$installedApps` contains not only the app objects but additionally warning messages that were written to output.
Executing the following statement in a PowerShell window also outputs the warning messages and app objects:
```
Get-NavContainerAppInfo -containerName $containername -tenantSpecificProperties -sort DependenciesLast | Where-Object { $_.Name -ne "System Application" }
```

**Comment by freddydk** [comment]
If you are running as administrator, this doesn't happen.

I think the right fix is to replace the last line in Get-NavContainerAppInfo from this:
    } -ArgumentList $args, $sort 
to this
    } -ArgumentList $args, $sort | Where-Object {$_ -isnot [System.String]}

That should get rid of the license warnings.

**Comment by freddydk** [comment]
This fix is shipped in 0.6.3.2
